packages/components/src/components/radio-group/radio-group.ts

```diff
@@ -45,28 +44,33 @@ import type SdRadioButton from '../../_components/radio-button/radio-button';
 export default class SdRadioGroup extends SolidElement implements SolidFormControl {
   static dependencies = { 'sd-button-group': SdButtonGroup };
 
-  protected readonly formControlController = new FormControlController(this);
-  private readonly hasSlotController = new HasSlotController(this, 'label', 'error-text');
+  protected readonly formControlController: FormControlController = new FormControlController(this);
   private customValidityMessage = '';
   private validationTimeout: number;
 
   @query('slot:not([name])') defaultSlot: HTMLSlotElement;
   @query('#validation-input') validationInput: HTMLInputElement;
+  @query('#error-message') errorMessage: HTMLDivElement;
 
   @state() private hasButtonGroup = false;
   @state() defaultValue = '';
 
-  /**  A Boolean attribute which, if present, marks the radio valid or invalid. Please note that 'invalid' can only be used in conjunction with 'this.required'.  */
-  @state() private invalid = false;
-
-  /** The radio groups's error text. Use to display an error message below the component. Please note that 'error-text' can only be used in conjunction with 'this.required' and 'this.invalid'.  */
-  @state() private errorText = '';
+  /**
+   * Indicates whether or not the user input is valid after the user has interacted with the component. These states are activated when the attribute "data-user-valid" or "data-user-invalid" are set on the component via the form controller. They are different than the native input validity state which is always either `true` or `false`.
+   */
+  @state() showInvalidStyle = false;
 
   /**
    * The radio group's label. Required for proper accessibility. If you need to display HTML, use the `label` slot
    * instead.
    */
-  @property() label = '';
+  @property() label = 'Label';
```

Comment:
Design explicitly state we should have default values for labels to enforce accessibility standards, I have applied this.